src/sparsebase/preprocess/preprocess.cc

```diff
@@ -1634,6 +1669,86 @@ IDType *MetisReorder<IDType, NNZType, ValueType>::GetReorderCSR(
 
 #endif
 
+template <typename IDType, typename NNZType, typename ValueType, typename FloatType>
+ReorderHeatmap<IDType, NNZType, ValueType, FloatType>::ReorderHeatmap(ReorderHeatmapParams params){
+  this->params_ =
+      std::make_unique<ReorderHeatmapParams>(params);
+  this->RegisterFunction(
+      {format::CSR<IDType, NNZType, ValueType>::get_id_static(), format::Array<IDType>::get_id_static()},
+      ReorderHeatmapCSRandArray);
+}
+template <typename IDType, typename NNZType, typename ValueType, typename FloatType>
+format::FormatOrderOne<FloatType>* ReorderHeatmap<IDType, NNZType, ValueType, FloatType>::Get(format::FormatOrderTwo<IDType, NNZType, ValueType> *format, format::FormatOrderOne<IDType>* permutation, std::vector<context::Context*> contexts, bool convert_input){
+  return this->Execute(this->params_.get(), contexts, convert_input, (format::Format*)format, (format::Format*)permutation);
+}
+
+template <typename IDType, typename NNZType, typename ValueType, typename FloatType>
+format::FormatOrderOne<FloatType>* ReorderHeatmap<IDType, NNZType, ValueType, FloatType>::ReorderHeatmapCSRandArray(std::vector<format::Format*> formats, PreprocessParams* poly_params){
+  auto csr = formats[0]->AsAbsolute<format::CSR<IDType, NNZType, ValueType>>();
+  auto order = formats[1]->AsAbsolute<format::Array<IDType>>()->get_vals();
+  auto* params = static_cast<ReorderHeatmapParams*>(poly_params);
+  int b = params->num_parts;
+  auto n = csr->get_dimensions()[0];
+  auto row_ptr = csr->get_row_ptr();
+  auto adj = csr->get_col();
+  long long max_bw = 0;
+  double mean_bw = 0;
```

Comment:
We must check if num parts is larger than number of rows

Reply:
Facts.

Reply:
Good point.